DANS-KNAW/dans-datastation-tools
# datastation-module-versions

NO JIRA

# Description of changes
Added a command `datastation-get-component-versions` which prints a table of the installed component versions, e.g.,

```
[vagrant@archaeology dans-datastation-tools]$ poetry run datastation-get-component-versions
                         Data Station Component Versions                         
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Component                               ┃ Version                             ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ dans.knaw.nl-dd-validate-dans-bag       │ 0.33.0                              │
│ dans.knaw.nl-dd-ingest-flow             │ 0.57.0                              │
│ dans.knaw.nl-dd-vault-metadata          │ 2.2.0                               │
│ dans.knaw.nl-dans-schema                │ 0.13.0                              │
│ dans.knaw.nl-dd-verify-migration        │ 2.5.0                               │
│ dans.knaw.nl-dd-virus-scan              │ 0.8.0                               │
│ dans.knaw.nl-dd-sword2                  │ 0.14.0                              │
│ dans.knaw.nl-dd-dataverse-authenticator │ 0.10.0                              │
│ dans.knaw.nl-dd-verify-dataset          │ 0.12.0                              │
│ dans.knaw.nl-dd-manage-deposit          │ 0.4.0                               │
│ dataverse                               │ 5.14 build DANS-DataStation-PATCH-4 │
│ payara                                  │ 5.2022.5                            │
└─────────────────────────────────────────┴─────────────────────────────────────┘


```
Handy for application managers to get a quick overview of the current baseline.

# How to test
Check out the branch on a vagrant box or a server,  do `poetry install` and then `poetry run`.  Make sure you have the following section in your config file:

```yaml
version_info:
  dans_rpm_module_prefix: 'dans.knaw.nl-'
  dataverse_application_path: '/var/lib/payara5/glassfish/domains/domain1/applications/dataverse/'
  payara_install_path: '/usr/local/payara5'


```



# Notify

@DANS-KNAW/dataversedans
